### PR TITLE
Feat: Implement openURL

### PR DESCRIPTION
**Pull request by Al0den**
Fix for #310

### CODE REVIEW

File: lua/copilot/api.lua [hunk 1]
```diff
@@ -227,12 +227,18 @@ mod.handlers = {
   statusNotification = status.handlers.statusNotification,
   ---@param result copilot_open_url_data
   ["copilot/openURL"] = function(_, result)
-    vim.api.nvim_echo({
-      { "copilot/openURL" },
-      { vim.inspect({ _, result }) },
-      { "\n", "NONE" },
-    }, true, {})
-    error("not implemented: copilot.api.handlers['copilot/openURL']")
+    if package.config:sub(1, 1) == '/' then
```
[the remark this answers is not in the document]
Comment (Al0den) [reply lua/copilot/api.lua:230]: image
No vim.ui.open on 
image

[the remark this answers is not in the document]
Comment (tmillr) [reply lua/copilot/api.lua:230]: That's very strange. It's in the docs `:h vim.ui.open()` and it works for me and im on 0.10.1 as well. You need to run `:version` from within nvim to check the version.

https://neovim.io/doc/user/lua.html#_lua-module:-vim.ui

**Comment by zbirenbaum** [comment]
According to neovim changelogs vim.ui.open has existed since 0.10. Lets use @tmillr suggestion and do something like this:

```
  ["copilot/openURL"] = function(_, result)
    local success, _ = pcall(vim.ui.open, result.target)
    if not success then
      if vim.ui.open ~= nil then
        vim.api.nvim_echo({
            { "copilot/openURL" },
            { vim.inspect({ _, result }) },
            { "\n", "NONE" },
        }, true, {})
        error("Unsupported OS: vim.ui.open exists but failed to execute.")
      else
        vim.api.nvim_echo({
            { "copilot/openURL" },
            { vim.inspect({ _, result }) },
            { "\n", "NONE" },
        }, true, {})
        error("Unsupported Version: vim.ui.open requires Neovim > 0.10")
      end
    end
  end
}
```

**Comment by Al0den** [comment]
Super weird, was on 0.10 and didn't have it.
Anyways, changed the commit to include this, thanks!

**Pull request merged**